cnxarchive/utils/ident_hash.py

```diff
@@ -82,3 +81,186 @@ def join_ident_hash(id, version):
     if version:
         join_args.append(version)
     return HASH_CHAR.join(join_args)
+
+
+class CNXHash(uuid.UUID):
```

Comment:
What you said here is valid but doesn't mean we have to subclass `uuid.UUID`, we can always write a class that uses `uuid.UUID` (like storing a uuid object as an attribute).

The other thing I've noticed is that if we have so many class methods, we have to think about whether this should be a class.  What advantages do we have making this a class vs having functions?